### PR TITLE
fix: pass context for isDefined and custom validators

### PR DESCRIPTION
**Pull request by chiangf**
Context wasn't being passed in for isDefined and custom validators.

https://github.com/typestack/class-validator/issues/292

### CODE REVIEW

**Comment by chiangf** [comment]
@amirsaber Would you be able to help review this? Thanks!

**Comment by devniel** [comment]
just discovered this problem today, thanks for the PR !

**Comment by taboulot** [comment]
When this will be merged and available on NPM :) ?

**Comment by dolsem** [comment]
Any update on this?

**Comment by geersch** [comment]
Any update on this?

**Comment by chiangf** [comment]
I'm happy to resolve the conflicts, if someone is willing to review?

**Comment by vlapo** [comment]
Sure. Please resolve conflicts, cover changes with tests and I am happy to review this.

**Comment by chiangf** [comment]
@vlapo should be ready for review, thanks!

File: test/functional/custom-decorators.spec.ts [hunk 1]
```diff
@@ -87,6 +88,16 @@ describe("custom decorators", function() {
                 errors[0].constraints.should.be.eql({ isLongerThan: "firstName must be longer then lastName. Given value: Li" });
             });
         });
+
+        it("should include context", function() {
```
Comment (vlapo) [inline test/functional/custom-decorators.spec.ts]: Could you please move this test to `test/functional/validation-options.spec.ts` where all tests with context feature are?

Comment (chiangf) [reply test/functional/custom-decorators.spec.ts]: 👍 sounds good! Done in https://github.com/typestack/class-validator/pull/296/commits/e68b65eb37c11af4fa428aafde5182c00b787908

**Comment by vlapo** [comment]
Thank you for your patient and perseverance.

**Pull request merged**